j5/backends/hardware/sr/v4/serial/power_board.py

```diff
@@ -0,0 +1,298 @@
+                raise DeviceMissingSerialNumberError(
+                    "Found power board-like device without serial number. "
+                    f"The power board is likely to be damaged: {port.usb_info()}",
+                )
+            else:
+                boards.add(
+                    PowerBoard(
+                        port.serial_number,
+                        cast(
+                            Backend,
+                            cls(port.device),
+                        ),
+                    ),
+                )
+
+        return boards
+
+    def __init__(
+            self,
+            serial_port: str,
+    ) -> None:
+        super().__init__(
+            serial_port=serial_port,
+            baud=115200,
+        )
+        self._lock = threading.Lock()
+        self._led_states: Dict[int, bool] = {
+            i: False
+            for i in range(2)
+        }
+        self._check_firmware_version_supported(
+            self.firmware_version, MINIMUM_MAJOR_VERSION,
+        )
+        self.reset_board()
+
+    def get_power_output_enabled(self, identifier: int) -> bool:
+        """
+        Get whether a power output is enabled.
+
+        :param identifier: power output to fetch status of.
+        :returns: status of the power output.
+        :raises CommunicationError: Invalid response received.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            response = self.query(f"OUT:{identifier}:GET?")
+            if response == "0":
+                return False
+            elif response == "1":
+                return True
+            else:
+                raise CommunicationError(
+                    f"Power Board returned an invalid response: {response}")
+        else:
+            raise ValueError(f"{identifier!r} is not a valid power output identifier")
+
+    def set_power_output_enabled(
+        self, identifier: int, enabled: bool,
+    ) -> None:
+        """
+        Set whether a power output is enabled.
+
+        :param identifier: power output to enable / disable
+        :param enabled: status of the power output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            state = "1" if enabled else "0"
+            self.request(f"OUT:{identifier}:SET:{state}")
+        else:
+            raise ValueError(f"{identifier!r} is not a valid power output identifier")
+
+    def get_power_output_current(self, identifier: int) -> float:
+        """
+        Get the current being drawn on a power output, in amperes.
+
+        :param identifier: power output to fetch current of.
+        :returns: current of the output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            response = self.query(f"OUT:{identifier}:I?")
+            return float(response) / 1000
```

Comment:
Just realised that we aren't handling float throwing value errors throughout this file.